src/app/admin/components/courthouses/courthouses.component.ts

```diff
@@ -1,28 +1,56 @@
 import { CourthouseSearchFormValues } from '@admin-types/courthouses/courthouse-search-form-values.type';
-import { Component } from '@angular/core';
+import { Courthouse } from '@admin-types/courthouses/courthouse.type';
+import { CommonModule } from '@angular/common';
+import { Component, inject } from '@angular/core';
 import { GovukHeadingComponent } from '@common/govuk-heading/govuk-heading.component';
-import { BehaviorSubject, Subject } from 'rxjs';
+import { CourthouseService } from '@services/courthouses/courthouses.service';
+import { BehaviorSubject, Observable, Subject, combineLatest, of, switchMap, tap } from 'rxjs';
 import { CourthouseSearchFormComponent } from './courthouse-search-form/courthouse-search-form.component';
+import { CourthouseSearchResultsComponent } from './courthouse-search-results/courthouse-search-results.component';
 
 @Component({
   selector: 'app-courthouses',
   standalone: true,
   templateUrl: './courthouses.component.html',
   styleUrl: './courthouses.component.scss',
-  imports: [GovukHeadingComponent, CourthouseSearchFormComponent],
+  imports: [CommonModule, GovukHeadingComponent, CourthouseSearchFormComponent, CourthouseSearchResultsComponent],
 })
 export class CourthousesComponent {
+  courthouseService = inject(CourthouseService);
+
   search$ = new Subject<CourthouseSearchFormValues | null>();
   loading$ = new Subject<boolean>();
   isSubmitted$ = new BehaviorSubject<boolean>(false);
 
+  courthouses$ = this.courthouseService.getCourthousesWithRegions();
+
+  results$: Observable<Courthouse[]> = combineLatest([this.search$, this.isSubmitted$, this.courthouses$]).pipe(
+    tap(() => this.startLoading()),
+    switchMap(([values, isSubmitted, courthouses]) => {
+      if (!values || !isSubmitted || !courthouses) {
+        return of([]);
+      }
+      // Directly use the emitted courthouses array for filtering
+      return this.courthouseService.searchCourthouses(courthouses, values);
+    }),
+    tap(() => this.stopLoading())
+  );
+
+  startLoading() {
+    this.loading$.next(true);
+  }
+
+  stopLoading() {
+    this.loading$.next(false);
+  }
+
   onSubmit(values: CourthouseSearchFormValues) {
     this.isSubmitted$.next(true);
```

Comment:
Check this isn't causing duplicate subsequent requests as this is what was causing issues with User search

Reply:
The combineLatest update to include courthouses$ has prevented subsequent API calls